docs/hugo/content/architecture-decision-records/ADR-2022-02-Backward-Compatibility.md

```diff
@@ -1,17 +1,88 @@
 
-We don't want to maroon existing users of Azure Service Operator (ASO) by making this a breaking change.
+## Option 1: Do nothing
 
-Cross version compatibility between ARM versions of a resource (that is, between `v1alpha1api20180101` and `v1alpha1api20200101`) is already implemented and well tested, so we want to leverage that existing infrastructure.
+Leave all generated resources using the alpha prefix that is currently in use.
+
+Pros
+
+* Easy to implement (nothing to do) and quick to deliver
+
+Cons
+
+* Introduces an inconsistency between the status of ASO (beta) and the labelling of the resources (alpha).
+* Might be perceived by some users as indicating these particular resources are still in alpha, and therefore pose significant a barrier to adoption of ASO.
+
+## Option 2: Move all resources to beta
+
+Change the generated prefix for all resources
+
+Pros
+
+* Easy to implement (change the prefix, update affected tests, and regenerate all files), and quick to deliver
+
+Cons
+
+* Existing clusters with custom resources (CRs) reliant on ASO would break because ASO would no longer support the older version.  
+  Mitigation would require deleting the existing resource and creating a new CR with an updated version for the resource.
+* Existing automated deployments (e.g. as used by teams with a GitOpts model or other forms of continuous deployment) would break because the version specified in the existing YAML files would no longer exist.  
+  Mitigation would require modification of the deployment pipelines to change the version of the resources being deployed.
+* Existing manual deployments would also break because the version specified in the existing YAML files would no longer exist.
+  Mitigation would require modification of the YAML files being deployed to change the version of the resources being deployed.
+
+## Option 3a: In-place upgrade for existing resources
+
+Change the generated prefix to `v1beta` for all resources, but retain the alpha storage variants of each existing resource. This would allow an upgraded ASO installation to promote any existing resources to the new storage version.
+
+Pros
+
+* Moderate effort to implement (requires a new pipeline stage to create the alpha storage variants for compatibility with older versions).
+* Existing clusters with custom resources reliant on ASO would continue to run because ASO would support conversion from the old hub version to the new.
+
+Cons
+
+* Existing automated deployments (e.g. as used by teams with a GitOpts model or other forms of continuous deployment) would break because the version specified in the existing YAML files would no longer exist.  
+  Mitigation would require modification of the deployment pipelines to change the version of the resources being deployed.
+* Existing manual deployments would also break because the version specified in the existing YAML files would no longer exist.
```

Comment:
Might be clearer to say users attempting to update existing CRs (by reapplying an updated YAML). I think the word "deployments" is ambiguous because there is a Kubernetes resource Deployments (that doesn't come into play here at all). 

Since this is basically the same thing as the first bullet you could also combine them into something like:
"Any attempt to apply old v1alpha1 YAML will fail, as it no longer exists. This includes attempting to manually apply a YAML using `kubectl apply -f` or through some other mechanism where YAMLs are continuously applied, as in GitOps style workflows"

Reply:
Good phrasing, I'll use that.